scripts/windows/setup/IotEdgeSecurityDaemon.ps1

```diff
@@ -143,9 +155,19 @@ function Initialize-IoTEdge {
         [Switch] $External,
 
         # The device connection string.
-        [Parameter(Mandatory = $true, ParameterSetName = 'Manual')]
```

Comment:
This is going to cause more problems like https://github.com/Azure/iotedge/issues/1486

I think at this stage it's worth making several more finer-grained parameter sets. `Manual_DCS`, `Manual_X509`, `DPS_TPM`, `DPS_SymmetricKey` and `DPS_X509`. Then set `Mandatory` for the sets that need them.

Reply:
Thanks for this feedback and your help. I have cleaned this up per your comments and ensured backward compat as well.